packages/cadl-ranch-specs/http/azure/core/basic/mockapi.ts

```diff
@@ -74,14 +74,22 @@ Scenarios.Azure_Core_Basic_list = passOnSuccess(
   }),
 );
 
-Scenarios.Azure_Core_Basic_listWithPage = passOnSuccess(
+Scenarios.Azure_Core_Basic_listWithPage = passOnSuccess([
   mockapi.get("/azure/core/basic/page", (req) => {
     const responseBody = {
       value: [validUser],
+      nextLink: '/azure/core/basic/page/nextLink?%24skiptoken={"bar"%3a"foo"}',
     };
     return { status: 200, body: json(responseBody) };
   }),
-);
+  mockapi.get("/azure/core/basic/page/nextLink", (req) => {
+    req.expect.containsQueryParam("$skiptoken", '{"bar":"foo"}');
+    const responseBody = {
+      value: [],
```

Comment:
I am a bit uncomfortable with the empty in 2nd page. Assume all SDK pagination can handle it.

Reply:
autorest.testserver has [similar testcase](https://github.com/Azure/autorest.testserver/blob/b6195830209d783dcfef44803f4748f35c1241e0/src/test-routes/paging.ts#L65-L92) long time ago so I infer all SDK has supported it.

Reply:
It also was originally added to autorest.testserver because a service had this behavior, I can't remember the service though but Python failed on it so we added it as a test

Reply:
Got it. I will update a new check if we support next link contains `{}`

Reply:
Personally, I'd prefer unit test in autorest.python for this particular case of `{}`.

"{} in nextLink" is a too limited case. One can then have "= in nextLink", ", in nextLink", "% in nextLink"... It does not worth it.

If we want to add to cadl-ranch, I'd prefer it cover more characters like `=` etc.